test/upgrade/fleet.crt

```diff
@@ -0,0 +1,28 @@
+-----BEGIN CERTIFICATE-----
```

Comment:
The way the networking was done in the docker-compose.yml file for these tests is a little different. When using docker-compose, you can use the service name as the host. osquery connects to fleet using `fleet` and doesn't rely on the port exposed on the host running docker-compose. I had to generate new certs that have the common name set to `fleet`. I also wanted to make these tests fairly self contained, so that we do not need to worry about breaking these tests when making changes elsewhere.

The advantages of doing it this way are

- No port conflicts. Free ports are automatically assigned by docker
- Don't need to change the osquery config when switching from fleet a to fleet b
- Can potentially run multiple tests in parallel as long as there are enough resources available
- Tests are self contained. Everything needed is in test/upgrade. Don't need to worry about breaking changes elsewhere.

Reply:
makes sense, thank you very much for the details